python/ray/data/iterator.py

```diff
@@ -852,6 +861,35 @@ def generator():
         )
         return dataset.with_options(options)
 
+    def materialize(self) -> "MaterializedDataset":
+        """Execute and materialize this data iterator into object store memory.
+
+        This method will fully consume the iterator and return its contents as
+        a `MaterializedDataset` view.
+        """
+
+        from ray.data.dataset import MaterializedDataset
+
+        block_iter, stats, owned_by_consumer = self._to_block_iterator()
+
+        block_refs_and_metadata = list(block_iter)
```

Comment:
Can we add a comment that this would trigger the execution and materialize all blocks of iterator? Want to make it obvious for people to read in the future.